api/datastore/sql/sql.go

```diff
@@ -362,7 +362,16 @@ func (ds *sqlStore) GetApp(ctx context.Context, name string) (*models.App, error
 // GetApps retrieves an array of apps according to a specific filter.
 func (ds *sqlStore) GetApps(ctx context.Context, filter *models.AppFilter) ([]*models.App, error) {
 	res := []*models.App{}
-	query, args := buildFilterAppQuery(filter)
+	if filter.NameIn != nil && len(filter.NameIn) == 0 { // this basically makes sure it doesn't return ALL apps
```

Comment:
this seems specific to the caller that was worked on in conjunction with this. passing a nil filter.NameIn returns all apps so this is kind of confusing here to change behavior based on the nil-ness of an empty list in either case.

Reply:
Ya, I agree, didn't really know a good way to handle this. By default it's nil and would return all apps, as expected. If someone sets it, even if it's empty, then assuming it's supposed to be used.  I could see extension developers misusing this though and causing a gaping security hole (although they'd have to set something that let's the filter know not to return all regardless). Open to suggestions.